src/metrics/timingMetric.js

```diff
@@ -0,0 +1,43 @@
+import {VALID_TYPES} from './metrics';
+
+function typeOrError(type) {
+  if (VALID_TYPES.includes(type)) {
+    return type;
+  }
+  throw new Error('invalid type for a metric supplied');
+}
+
+export default function newMetric(metricType, {fields, tags} = {
+  fields: [],
+  tags: []
+}, stop = undefined, start = undefined) {
+  const type = typeOrError(metricType);
+  const startTime = start || new Date();
+
+  return {
+    addField(key, value) {
+      fields[key] = value;
```

Comment:
Since fields and tags are passed by reference this also alters any original `fields` object passed into the metric. Might be an idea to make them specific to the returned object instead.